concrete-core/src/backends/core/implementation/engines/glwe_ciphertext_conversion.rs

```diff
@@ -0,0 +1,182 @@
+    /// let ciphertext =
+    ///     engine.encrypt_glwe_ciphertext(&key, &plaintext, noise)?;
+    ///
+    /// // Then we convert it to the Fourier domain.
+    /// let fourier_ciphertext: FourierGlweCiphertext32 = engine.convert_glwe_ciphertext(ciphertext)?;
+    /// #
+    /// assert_eq!(fourier_ciphertext.glwe_dimension(), glwe_dimension);
+    /// assert_eq!(fourier_ciphertext.polynomial_size(), polynomial_size);
+    /// assert_eq!(fourier_ciphertext.decomposition_base_log(), base_log);
+    /// assert_eq!(fourier_ciphertext.decomposition_level_count(), level);
+    ///
+    /// engine.destroy(key)?;
+    /// engine.destroy(plaintext)?;
+    /// engine.destroy(ciphertext)?;
+    /// engine.destroy(fourier_ciphertext)?;
+    /// #
+    /// # Ok(())
+    /// # }
+    /// ```
+    fn convert_glwe_ciphertext(
+        &mut self,
+        input: &GlweCiphertext32,
+    ) -> Result<FourierGlweCiphertext32, GlweCiphertextConversionError<Self::EngineError>> {
+        if !ALLOWED_POLY_SIZE.contains(&input.polynomial_size().0) {
+            return Err(GlweCiphertextConversionError::from(
+                CoreError::UnsupportedPolynomialSize,
+            ));
+        }
+
+        Ok(unsafe { self.convert_glwe_ciphertext_unchecked(input) })
+    }
+
+    unsafe fn convert_glwe_ciphertext_unchecked(
+        &mut self,
+        input: &GlweCiphertext32,
+    ) -> FourierGlweCiphertext32 {
+        let mut output = FourierGlweCiphertext::allocate(
+            Complex64::new(0., 0.),
+            input.polynomial_size(),
+            input.glwe_dimension().to_glwe_size(),
+        );
+        let buffers = self.get_fourier_u32_buffer(
+            input.polynomial_size(),
+            input.glwe_dimension().to_glwe_size(),
+        );
+        output.fill_with_forward_fourier(&input.0, buffers);
+        FourierGlweCiphertext32(output)
+    }
+}
+
+/// # Description:
+/// Implementation of [`GlweCiphertextConversionEngine`] for [`CoreEngine`] that operates on
+/// 64 bits integers. It converts a GLWE ciphertext from the standard to the Fourier domain.
+impl GlweCiphertextConversionEngine<GlweCiphertext64, FourierGlweCiphertext64> for CoreEngine {
+    /// # Example
+    /// ```
+    /// use concrete_commons::dispersion::Variance;
+    /// use concrete_commons::parameters::{
+    ///     DecompositionBaseLog, DecompositionLevelCount, GlweDimension, LweDimension, PolynomialSize,
+    /// };
+    /// use concrete_core::prelude::*;
+    /// # use std::error::Error;
+    ///
+    /// # fn main() -> Result<(), Box<dyn Error>> {
+    /// // DISCLAIMER: the parameters used here are only for test purpose, and are not secure.
+    /// let glwe_dimension = GlweDimension(2);
+    /// let polynomial_size = PolynomialSize(256);
+    /// let level = DecompositionLevelCount(1);
+    /// let base_log = DecompositionBaseLog(4);
+    /// // Here a hard-set encoding is applied (shift by 50 bits)
+    /// let input = vec![3_u64 << 20; 3];
```

Comment:
Here the input should be a vec of 256 elements (not 3). The two doc tests should be exactly the same except for the integer representation :smiley: